### ISSUE TITLE
error: Invalid value for '<binary>': /input is not a file.

### ISSUE THREAD

**Issue by frakman1** [fkie-cad/cwe_checker#218]
Due to recent changes to this docker image, the previous method of running this tool is now failing with this error:

```
error: Invalid value for '<binary>': /input is not a file.
```

To make matters worse, there is no history of tags on dockerhub to use a previous working version.

I found a workaround by overriding the entrypoint. Trickier to do within a Gitlab (`gitlab-ci.yml`) pipeline but also possible.
I also found that `cwe_checker` no longer works if called within docker and the full path must now be specified (`/home/cwe/cwe_checker`)

It would be nice if:
1- Changes were not breaking or backward incompatible.
2- If they have to be, then at least a previous tag be made available to maintain historical behavior.

However, since I wish to use the latest iteration of this tool, it would be nice to be able to use it consistently.

For context, this is my current gitlab job specification:

```yaml
.cwechecker: &cwechecker
  stage: cwechecker
  image: fkiecad/cwe_checker:stable
  allow_failure: true
  before_script:
    - <removed>
  script:
    - set +e
    - cd xxx/rootfs/
    - for filename in $(cut -d':' -f1 <<< "$(find . -type f -exec file {} + | grep ELF)") ; do echo "#######" ; ls $filename ; echo "#######" ; cwe_checker "$filename" ; done | tee cwechecker-results.txt
    - cd -
    - cp xxx/rootfs/cwechecker-results.txt output/
    - set -e
  artifacts:
    name: <removed>
    paths:
      - output/cwechecker-results.txt
    expire_in: 3 days
```

For completeness, the workaround was adding:
```yaml
entrypoint: [""]
```
under the line with the `image:` keyword

**Comment by Enkelmann** [comment]
I assume the problem you encountered is that the preferred way to use the docker image changed from
```
docker run --rm -v /PATH/TO/BINARY:/input fkiecad/cwe_checker cwe_checker /input
```
in the previous stable version to
```
docker run --rm -v /PATH/TO/BINARY:/input fkiecad/cwe_checker /input
```
in the latest stable version. There was a warning about this change (and other breaking changes between v0.4 and v0.5) on the Readme.md for the master branch for some months, but I removed it when it became the new stable version.

Please keep in mind that the cwe_checker still has not hit version 1.0 ! And although we try to keep breaking changes to a minimum, they can and will happen between version jumps (e.g. v0.4 to v0.5) if we deem them necessary. I am also hesitant to provide docker images older than the latest stable, since up to now it made more sense to force users to use the newest stable version because the improvements in each version step were considerable. But I will think about it for future stable versions! We cannot provide stronger stability guarantees until we think the cwe_checker is ready for a version 1.0.

I hope this answers your issue.

**Comment by frakman1** [comment]
Thank you for the insight.

I don't actually use this image via the conventional `docker run` syntax so a `README.md` update would not have raised any flags.

What happens in the Gitlab CICD is that the docker container is created (somehow by the framework) and commands from my CICD script are executed within the container. I only control the `script` portion of the above snippet of the gitlab specification I pasted and not how the docker image is run.

The error I got was this:

![image](https://user-images.githubusercontent.com/5826484/131364380-c7e3e2ff-c0bc-4e74-b714-913a5e66578a.png)

I suspected an ENTRYPOINT/CMD type change and confirmed that by looking at the Dockerfile history.
